tools/Custom/PSCmdletExtensions.cs

```diff
@@ -29,7 +29,7 @@ internal static string UnescapeString(this PSCmdlet cmdlet, string value)
             try
             {
                 var unescapedString = Uri.UnescapeDataString(value);
-                return value.EndsWith("'") ? unescapedString: unescapedString.Replace('+', ' ');
+                return (value.EndsWith("'") || value.EndsWith("')")) ? unescapedString: unescapedString.Replace('+', ' ');
```

Comment:
Is it possible to capture these scenarios in a test to prevent a regression in the future?

Reply:
This will be tricky because the extension method a cmdlet derived from ``PSCmdlet`` class and cmdlets derived from the PSCmdlet class in PowerShell cannot be invoked directly because they are designed to be used within the context of a PowerShell runtime environment.

For example in a unit test we will need to have something like the code below calling a cmdlet implementation.
```
        var cmdlet = new CommandA();
        cmdlet.Value = "('Khalayi+mzuri')";
        cmdlet.Invoke();
```
The above won't work because we will not have a PowerShell run time which ``PSmdlet`` class depends on.

Below is the cmdlet implementation which we hope it can use the extension method.
```
using System.Management.Automation;

public class CommandA : PSCmdlet
{
    [Parameter(Position = 0, Mandatory = true)]
    public string Value { get; set; }

    protected override void ProcessRecord()
    {
       WriteObject(this.UnescapeString(Value));
    }
}
```
This is one of the limitations we currently having when it comes to writing unit testing some an extension method expecting a Cmdlet class. 

Maybe you can give more suggestion on how to go about this.

Reply:
Thanks for extra context here @timayabi2020 

I believe it should still be possible to add a unit test library to the `tools` projects and have the unit tests target the specific method. Another alternative would be to refactor out the extension methods into separately testable components so that these methods simply call the testable methods. (All these are ideas thoughts outside the scope of this PR though, would you be willing to create a separate issue to investigate?)

Otherwise, we can resolve this on for now.